server/app/services/applicant/ApplicantService.java

```diff
@@ -465,34 +471,47 @@ CompletionStage<Application> submitApplication(
               Program applicationProgram = application.getProgram();
               ProgramDefinition programDefinition = applicationProgram.getProgramDefinition();
               String programName = programDefinition.adminName();
-
-              notifyProgramAdmins(applicantId, programId, application.id, programName);
-
-              CompletableFuture<Void> maybeNotifyTiSubmitterFuture =
-                  tiSubmitterEmail.isPresent()
-                      ? maybeNotifyTiSubmitter(
-                              tiSubmitterEmail.get(), applicantId, application.id, programName)
-                          .toCompletableFuture()
-                      : CompletableFuture.completedFuture(null);
-
               Optional<StatusDefinitions.Status> maybeDefaultStatus =
                   applicationProgram.getDefaultStatus();
-              CompletableFuture<Void> maybeUpdateStatusAndEmailApplicantFuture;
-              if (maybeDefaultStatus.isPresent()) {
-                StatusDefinitions.Status defaultStatus = maybeDefaultStatus.get();
-                maybeUpdateStatusAndEmailApplicantFuture =
-                    CompletableFuture.allOf(
-                        setApplicationStatus(application, defaultStatus).toCompletableFuture(),
-                        maybeNotifyApplicantWithStatus(
-                                applicantId, programDefinition, defaultStatus)
-                            .toCompletableFuture());
-              } else {
-                maybeUpdateStatusAndEmailApplicantFuture =
-                    maybeNotifyApplicant(applicantId, application.id, programName)
-                        .toCompletableFuture();
-              }
+              Optional<String> applicantEmail = applicantEmailFuture.join();
+
+              CompletableFuture<ApplicationEvent> maybeUpdateStatusFuture =
+                  maybeDefaultStatus
+                      .map(
+                          status -> setApplicationStatus(application, status).toCompletableFuture())
+                      .orElse(CompletableFuture.completedFuture(null));
+              CompletableFuture<Void> notifyProgramAdminsFuture =
+                  CompletableFuture.runAsync(
+                      () ->
+                          notifyProgramAdmins(applicantId, programId, application.id, programName));
+              CompletableFuture<Void> maybeNotifyTiSubmitterFuture =
+                  tiSubmitterEmail
+                      .map(
```

Comment:
I think `isPresent()` may be more readable:

```java
CompletableFuture<Void> maybeNotifyTiSubmitterFuture =
    tiSubmitterEmail.isPresent() ? 
            notifyTiSubmitter(
                    tiSubmitterEmail.get(),
                    applicantId,
                    application.id,
                    programName,
                    maybeDefaultStatus)
                .toCompletableFuture()
        : CompletableFuture.completedFuture(null);
```

Happy with either though. What do you think?

Reply:
SG, I'm happy to keep it as-is.